#**Integantes**

* Maria Camila Londoño Muñoz
* Diego Andrés Luna Paternina
* Andrés Esteban Marín Manco

#**Importar Librerías**

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px
import plotly.graph_objects as go

!pip install unidecode
import unidecode

!pip install pydeck
import pydeck as pdk

     |████████████████████████████████| 235 kB 7.8 MB/s 
     |████████████████████████████████| 4.3 MB 6.9 MB/s 
     |████████████████████████████████| 130 kB 45.1 MB/s 
     |████████████████████████████████| 130 kB 38.2 MB/s 
     |████████████████████████████████| 428 kB 43.5 MB/s 
     |████████████████████████████████| 793 kB 16.1 MB/s 
     |████████████████████████████████| 381 kB 44.0 MB/s 
  Attempting uninstall: tornado
    Found existing installation: tornado 5.1.1
    Uninstalling tornado-5.1.1:
      Successfully uninstalled tornado-5.1.1
  Attempting uninstall: prompt-toolkit
    Found existing installation: prompt-toolkit 1.0.18
    Uninstalling prompt-toolkit-1.0.18:
      Successfully uninstalled prompt-toolkit-1.0.18
  Attempting uninstall: jupyter-client
    Found existing installation: jupyter-client 5.3.5
    Uninstalling jupyter-client-5.3.5:
      Successfully uninstalled jupyter-client-5.3.5
  Attempting uninstall: ipython
    Found existing installation: ipyt

##**Funciones**

In [2]:
#Remover acentos y caracteres especiales
def quitar_acento(string):
    return unidecode.unidecode(string)

In [3]:
#Comprobar si los elementos dentro de una lista o las columnas de un DF están en otro DF
def comprobar_columnas(lista1, lista2):
  if len(lista1) == len(lista2):
    print('Los DataFrames tienen la misma cantidad de columnas.')
  else:
    print('Los DataFrames no tienen la misma cantidad de columnas.')
  
  not_in = []
   
  for i in lista1:
    if i not in lista2:
      not_in.append(i)

  for i in lista2:
    if i not in lista1:
      not_in.append(i)

  if len(not_in) == 0:
    print('Los DateFrames tienen las mismas columnas. \n')
  else:
    print(f'Las siguientes columnas no están en alguno de los DataFrames: {not_in} \n')

In [4]:
#Añadir numero según el día de la semana
def num_dia_sem(fila):
  if fila == 'Lunes':
    v = 1
  elif fila == 'Martes':
    v = 2
  elif fila == 'Miércoles':
    v = 3
  elif fila == 'Jueves':
    v = 4
  elif fila == 'Viernes':
    v = 5
  elif fila == 'Sábado':
    v = 6
  else:
    v = 7
  return v


##**Acceder a Drive**

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#**Importar y limpiar bases de datos**

##**Importar BD**

In [50]:
#Leer base de datos que permitira asociar los codigos de las bases de datos con el nombre de las ciudades
#y resto de categorias que estan representadas por números
dicc = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/diccionario_accidentes_transito.xlsx')

#Importar bases de datos sobre fallecidos y lesionados desde el 2015 hasta el 2020
fl_2015 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2015.xlsx')
fl_2016 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2016.xlsx')
fl_2017 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2017.xlsx')
fl_2018 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2018.xlsx')
fl_2019 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2019.xlsx')
fl_2020 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/fallecidos_lesionados_2020.xlsx')

#Importar bases de datos sobre hechos de transito desde el 2015 hasta el 2020
ht_2015 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2015.xlsx')
ht_2016 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2016.xlsx')
ht_2017 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2017.xlsx')
ht_2018 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2018.xlsx')
ht_2019 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2019.xlsx')
ht_2020 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/hechos_transito_2020.xlsx')

#Importar bases de datos sobre vehiculos involucrados desde el 2015 hasta el 2020
vi_2015 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2015.xlsx')
vi_2016 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2016.xlsx')
vi_2017 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2017.xlsx')
vi_2018 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2018.xlsx')
vi_2019 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2019.xlsx')
vi_2020 = pd.read_excel('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/vehiculos_involucrados_2020.xlsx')

#Base de datos con la geolocalizacion de las ciudades de Guatemala (mas adelante presentar mapas)
glc = pd.read_csv('/content/drive/MyDrive/Analítica I/Trabajo Final/Accidentes de Transito/geolocalizaciones_ciudades.csv')

/usr/local/lib/python3.7/dist-packages/openpyxl/worksheet/_reader.py:312: UserWarning:

Unknown extension is not supported and will be removed



##**Concatenar las BD de las misma tematica**

###**fl**

In [51]:
###Se concatenara todas las bd de fl en todos los años

#Se pasa el nombre de las columnas a minusculas y se qitan los acentos para mejorar la coincidencia
#con las columnas de las otras bases de datos con la misma tematica
fl_2015.columns = map(quitar_acento, fl_2015.columns.str.lower())
fl_2016.columns = map(quitar_acento, fl_2016.columns.str.lower())
fl_2017.columns = map(quitar_acento, fl_2017.columns.str.lower())
fl_2018.columns = map(quitar_acento, fl_2018.columns.str.lower())
fl_2019.columns = map(quitar_acento, fl_2019.columns.str.lower())
fl_2020.columns = map(quitar_acento, fl_2020.columns.str.lower())

#Se verifican la cantidad de columnas de cada fl
print(len(fl_2015.columns))
print(len(fl_2016.columns))
print(len(fl_2017.columns))
print(len(fl_2018.columns))
print(len(fl_2019.columns))
print(len(fl_2020.columns))

comprobar_columnas(fl_2015.columns, fl_2016.columns)
comprobar_columnas(fl_2016.columns, fl_2017.columns)
comprobar_columnas(fl_2017.columns, fl_2018.columns)
comprobar_columnas(fl_2018.columns, fl_2019.columns)
comprobar_columnas(fl_2019.columns, fl_2020.columns)

#el atributo 'area_geo_ocu' no se encuentra en alguno de los fl, arrojara nulos al hacer el merge.
#Se debe eliminar de cada fl o hacerlo despues de unir todos los fl con el concat

26
26
25
25
25
25
Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames no tienen la misma cantidad de columnas.
Las siguientes columnas no están en alguno de los DataFrames: ['area_geo_ocu'] 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 



In [52]:
###Se concatenara todas las bd de fl en todos los años

fl = pd.concat([fl_2015,fl_2016,fl_2017,fl_2018,fl_2019,fl_2020])
#Se verifica la existencia de nulos. 2015 y 2016 tienen una columna de mas que el resto de años: 'area_geo_ocu' 
print(fl.isna().sum())
#Se elimina la columna 'area_geo_ocu', tiene 36838 nulos y por otro lado no se trabajara con esta columna
fl = fl.drop(columns='area_geo_ocu')
print('Ahora La cantidad de nulos es', fl.isna().sum().sum()) #Verifica la cantidad de nulos de nuevo en todo el fl

num_corre                0
ano_ocu                  0
mes_ocu                  0
dia_ocu                  0
dia_sem_ocu              0
hora_ocu                 0
g_hora                   0
g_hora_5                 0
depto_ocu                0
mupio_ocu                0
area_geo_ocu         36838
zona_ocu                 0
sexo_per                 0
edad_per                 0
mayor_menor              0
g_edad_80ymas            0
g_edad_60ymas            0
edad_quinquenales        0
fall_les                 0
int_o_noint              0
tipo_veh                 0
marca_veh                0
color_veh                0
modelo_veh               0
g_modelo_veh             0
tipo_eve                 0
dtype: int64
Ahora La cantidad de nulos es 0


###**ht**

In [53]:
#Se pasa el nombre de las columnas a minusculas y se qitan los acentos para mejorar la coincidencia
#con las columnas de las otras bases de datos con la misma tematica
ht_2015.columns = map(quitar_acento, ht_2015.columns.str.lower())
ht_2016.columns = map(quitar_acento, ht_2016.columns.str.lower())
ht_2017.columns = map(quitar_acento, ht_2017.columns.str.lower())
ht_2018.columns = map(quitar_acento, ht_2018.columns.str.lower())
ht_2019.columns = map(quitar_acento, ht_2019.columns.str.lower())
ht_2020.columns = map(quitar_acento, ht_2020.columns.str.lower())

#Se verifican la cantidad de columnas de cada ht
print(len(ht_2015.columns))
print(len(ht_2016.columns))
print(len(ht_2017.columns))
print(len(ht_2018.columns))
print(len(ht_2019.columns))
print(len(ht_2020.columns))

comprobar_columnas(ht_2015.columns, ht_2016.columns)
comprobar_columnas(ht_2016.columns, ht_2017.columns)
comprobar_columnas(ht_2017.columns, ht_2018.columns)
comprobar_columnas(ht_2018.columns, ht_2019.columns)
comprobar_columnas(ht_2019.columns, ht_2020.columns)

25
18
17
17
17
17
Los DataFrames no tienen la misma cantidad de columnas.
Las siguientes columnas no están en alguno de los DataFrames: ['sexo_per', 'edad_per', 'mayor_menor', 'g_edad_80ymas', 'g_edad_60ymas', 'edad_quinquenales', 'estado_con'] 

Los DataFrames no tienen la misma cantidad de columnas.
Las siguientes columnas no están en alguno de los DataFrames: ['area_geo_ocu'] 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 



In [54]:
###Se concatenara todas las bd de ht en todos los años

#se concatenan las bd de la misma tematica en diferentes anios
ht = pd.concat([ht_2015,ht_2016,ht_2017,ht_2018,ht_2019,ht_2020])
#Se verifica la existencia de nulos. 2015 y 2016 tienen una columna de mas que el resto de años: 'area_geo_ocu' 
print(ht.isna().sum())
#Se elimina las columnas con nulos, dado que no tenian registros en alguno de los otros años en esa variable.
ht = ht.drop(columns=['area_geo_ocu', 'sexo_per', 'edad_per', 'mayor_menor', 'g_edad_80ymas', 'g_edad_60ymas', 'edad_quinquenales', 'estado_con'])
print('Ahora La cantidad de nulos es', ht.isna().sum().sum()) #Verifica la cantidad de nulos de nuevo en todo el ht

num_corre                0
ano_ocu                  0
mes_ocu                  0
dia_ocu                  0
dia_sem_ocu              0
hora_ocu                 0
g_hora                   0
g_hora_5                 0
depto_ocu                0
mupio_ocu                0
area_geo_ocu         25671
zona_ocu                 0
sexo_per             33635
edad_per             33635
mayor_menor          33635
g_edad_80ymas        33635
g_edad_60ymas        33635
edad_quinquenales    33635
estado_con           33635
tipo_veh                 0
marca_veh                0
color_veh                0
modelo_veh               0
g_modelo_veh             0
tipo_eve                 0
dtype: int64
Ahora La cantidad de nulos es 0


###**vi**

In [55]:
#Se pasa el nombre de las columnas a minusculas y se qitan los acentos para mejorar la coincidencia
#con las columnas de las otras bases de datos con la misma tematica
vi_2015.columns = map(quitar_acento, vi_2015.columns.str.lower())
vi_2016.columns = map(quitar_acento, vi_2016.columns.str.lower())
vi_2017.columns = map(quitar_acento, vi_2017.columns.str.lower())
vi_2018.columns = map(quitar_acento, vi_2018.columns.str.lower())
vi_2019.columns = map(quitar_acento, vi_2019.columns.str.lower())
vi_2020.columns = map(quitar_acento, vi_2020.columns.str.lower())

#Se verifican la cantidad de columnas de cada ht
print(len(vi_2015.columns))
print(len(vi_2016.columns))
print(len(vi_2017.columns))
print(len(vi_2018.columns))
print(len(vi_2019.columns))
print(len(vi_2020.columns))

comprobar_columnas(vi_2015.columns, vi_2016.columns) #La vi_2015 no tiene la columna del anio. Se aniadira esta columna: ano_ocu
comprobar_columnas(vi_2016.columns, vi_2017.columns)
comprobar_columnas(vi_2017.columns, vi_2018.columns)
comprobar_columnas(vi_2018.columns, vi_2019.columns)
comprobar_columnas(vi_2019.columns, vi_2020.columns)

24
25
24
24
24
24
Los DataFrames no tienen la misma cantidad de columnas.
Las siguientes columnas no están en alguno de los DataFrames: ['ano_ocu'] 

Los DataFrames no tienen la misma cantidad de columnas.
Las siguientes columnas no están en alguno de los DataFrames: ['area_geo_ocu'] 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 

Los DataFrames tienen la misma cantidad de columnas.
Los DateFrames tienen las mismas columnas. 



In [56]:
#Agregar columna ano_ocu
vi_2015['ano_ocu'] = 2015
#mover ano_ocu de la ultima a la primera columna
col = vi_2015.pop('ano_ocu')
vi_2015.insert(0, 'ano_ocu', col)

In [57]:
#se concatenan las bd de la misma tematica en diferentes anios
vi = pd.concat([vi_2015,vi_2016,vi_2017,vi_2018,vi_2019,vi_2020])
#Se verifica la existencia de nulos. 2015 y 2016 tienen una columna de mas que el resto de años: 'area_geo_ocu' 
print(vi.isna().sum())
#Se elimina la columna 'area_geo_ocu', tiene 39088 nulos y por otro lado no se trabajara con esta columna
vi = vi.drop(columns='area_geo_ocu')
print('Ahora La cantidad de nulos es', vi.isna().sum().sum()) #Verifica la cantidad de nulos de nuevo en todo el vi

ano_ocu                  0
num_corre                0
mes_ocu                  0
dia_ocu                  0
dia_sem_ocu              0
hora_ocu                 0
g_hora                   0
g_hora_5                 0
depto_ocu                0
mupio_ocu                0
area_geo_ocu         39088
zona_ocu                 0
sexo_per                 0
edad_per                 0
mayor_menor              0
g_edad_80ymas            0
g_edad_60ymas            0
edad_quinquenales        0
estado_con               0
tipo_veh                 0
marca_veh                0
color_veh                0
modelo_veh               0
g_modelo_veh             0
tipo_eve                 0
dtype: int64
Ahora La cantidad de nulos es 0


##**Tratar fechas**

###**Fechas de fl**

In [58]:
fl.head(5)

num_corre  ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0          1     2015        1        1            4        16       3   
1          2     2015        1        1            4        16       3   
2          3     2015        1        1            4        22       4   
3          4     2015        1        1            4         2       1   
4          5     2015        1        1            4         9       2   

   g_hora_5  depto_ocu  mupio_ocu  ...  g_edad_60ymas  edad_quinquenales  \
0         2          1        101  ...              2                  4   
1         2          1        101  ...              2                  4   
2         3          1        101  ...              5                  7   
3         1          1        101  ...              2                  4   
4         1          1        101  ...              3                  5   

   fall_les  int_o_noint  tipo_veh  marca_veh  color_veh  modelo_veh  \
0         2            1         4         21          5        2011   
1         2            1         4         21          5        2011   
2         2            1         4         21          5        9999   
3         2            1         4         21          1        2000   
4         1            2         4         40          5        9999   

   g_modelo_veh  tipo_eve  
0             5         1  
1             5         1  
2             6         1  
3             4         1  
4             6         2  

[5 rows x 25 columns]

In [59]:
print(fl['ano_ocu'].isnull().sum())
print(fl['mes_ocu'].isnull().sum())
print(fl['dia_ocu'].isnull().sum()) #no hay nulos en las fechas, se puede concatenar en una fecha completa en formato fecha

print(fl['ano_ocu'].value_counts())
print(fl['mes_ocu'].value_counts())
print(fl['dia_ocu'].value_counts()) #no se observan datos anormales en las fechas, meses o dias

0
0
0
2016    11668
2019    10664
2015    10397
2018     9407
2017     8625
2020     8142
Name: ano_ocu, dtype: int64
12    5891
3     5539
11    5035
4     4952
1     4942
2     4893
9     4802
8     4694
5     4688
7     4595
10    4588
6     4284
Name: mes_ocu, dtype: int64
1     2433
2     2126
16    2102
18    2101
15    2086
21    2048
28    2022
20    1991
24    1971
25    1967
22    1965
13    1958
17    1944
14    1940
4     1908
19    1905
29    1905
26    1860
11    1853
10    1836
8     1836
30    1832
12    1831
27    1820
3     1813
6     1793
5     1770
7     1708
23    1675
9     1633
31    1271
Name: dia_ocu, dtype: int64


In [60]:
fl['fecha'] = fl['ano_ocu'].map(str) + '/' + fl['mes_ocu'].map(str) + '/' + fl['dia_ocu'].map(str) #concatenar la fecha completa en la columna fecha
fl['fecha'] = pd.to_datetime(fl['fecha'], dayfirst= False) #convertir los datos de la columna fecha en formato fecha
print(fl['fecha'].isnull().unique())
fl[58700:58750]

[False]


num_corre  ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
7939       7940     2020       12        5            6        19       4   
7940       7941     2020       12        5            6        21       4   
7941       7942     2020       12        5            6         1       1   
7942       7943     2020       12        5            6        19       4   
7943       7944     2020       12        5            6         8       2   
7944       7945     2020       12        5            6        18       4   
7945       7946     2020       12        6            7        10       2   
7946       7947     2020       12        6            7         0       1   
7947       7948     2020       12        6            7        16       3   
7948       7949     2020       12        6            7        20       4   
7949       7950     2020       12        6            7        21       4   
7950       7951     2020       12        6            7        19       4   
7951       7952     2020       12        6            7        18       4   
7952       7953     2020       12        6            7        18       4   
7953       7954     2020       12        6            7         0       1   
7954       7955     2020       12        6            7        17       3   
7955       7956     2020       12        6            7        12       3   
7956       7957     2020       12        7            1         9       2   
7957       7958     2020       12        7            1        20       4   
7958       7959     2020       12        7            1        20       4   
7959       7960     2020       12        7            1        19       4   
7960       7961     2020       12        7            1        19       4   
7961       7962     2020       12        7            1        17       3   
7962       7963     2020       12        7            1        22       4   
7963       7964     2020       12        8            2        16       3   
7964       7965     2020       12        8            2        18       4   
7965       7966     2020       12        8            2        19       4   
7966       7967     2020       12        8            2        14       3   
7967       7968     2020       12        8            2        18       4   
7968       7969     2020       12        9            3        12       3   
7969       7970     2020       12        9            3        20       4   
7970       7971     2020       12       10            4        18       4   
7971       7972     2020       12       10            4         6       2   
7972       7973     2020       12       10            4         6       2   
7973       7974     2020       12       10            4        21       4   
7974       7975     2020       12       10            4        19       4   
7975       7976     2020       12       10            4        19       4   
7976       7977     2020       12       10            4        20       4   
7977       7978     2020       12       10            4        21       4   
7978       7979     2020       12       10            4        20       4   
7979       7980     2020       12       11            5        19       4   
7980       7981     2020       12       11            5        23       4   
7981       7982     2020       12       12            6        19       4   
7982       7983     2020       12       12            6        10       2   
7983       7984     2020       12       12            6         3       1   
7984       7985     2020       12       12            6        20       4   
7985       7986     2020       12       12            6        21       4   
7986       7987     2020       12       12            6        16       3   
7987       7988     2020       12       12            6        21       4   
7988       7989     2020       12       12            6        20       4   

      g_hora_5  depto_ocu  mupio_ocu  ...  edad_quinquenales  fall_les  \
7939

###**Fechas de ht**

In [61]:
ht.head(5)
ht= ht.astype(int)
ht

num_corre  ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0             1     2015        1        1            4        16       3   
1             2     2015        1        1            4        22       4   
2             3     2015        1        1            4         2       1   
3             4     2015        1        1            4         9       2   
4             5     2015        1        1            4         1       1   
...         ...      ...      ...      ...          ...       ...     ...   
6345       6346     2020       12       31            4        19       4   
6346       6347     2020       12       31            4        20       4   
6347       6348     2020       12       31            4        13       3   
6348       6349     2020       12       31            4        18       4   
6349       6350     2020       12       31            4        21       4   

      g_hora_5  depto_ocu  mupio_ocu  zona_ocu  tipo_veh  marca_veh  \
0            2          1        101         1         4         21   
1            3          1        101        16         4         21   
2            1          1        101         7         3         44   
3            1          1        101        12         4         40   
4            1          1        115         8         1         34   
...        ...        ...        ...       ...       ...        ...   
6345         3         16       1601        99        99        999   
6346         3          2        207        99         1        999   
6347         2         14       1420        99         4          2   
6348         3          5        502        99         4        168   
6349         3          5        501        99         4         27   

      color_veh  modelo_veh  g_modelo_veh  tipo_eve  
0             5        2011             5         1  
1             5        9999             6         1  
2             6        9999             6         1  
3             5        9999             6         2  
4             4        9999             6         2  
...         ...         ...           ...       ...  
6345         99        9999            99         5  
6346          8        1984             2         5  
6347          5        9999            99         5  
6348          1        2019             5         5  
6349          1        2013             5         5  

[40489 rows x 17 columns]

In [62]:
print(ht['ano_ocu'].isnull().sum())
print(ht['mes_ocu'].isnull().sum())
print(ht['dia_ocu'].isnull().sum()) #no hay nulos en las fechas, se puede concatenar en una fecha completa en formato fecha
print(ht['ano_ocu'].value_counts())
print(ht['mes_ocu'].value_counts())
print(ht['dia_ocu'].value_counts()) #no se observan datos anormales en las fechas, meses o dias

0
0
0
2016    7964
2019    7047
2015    6854
2018    6395
2020    6350
2017    5879
Name: ano_ocu, dtype: int64
12    4104
3     3644
11    3557
2     3504
1     3390
10    3388
8     3299
9     3238
7     3198
4     3189
5     3029
6     2949
Name: mes_ocu, dtype: int64
1     1598
15    1454
16    1437
2     1426
18    1405
17    1390
20    1375
12    1356
14    1345
22    1342
3     1338
24    1327
21    1326
13    1315
28    1313
8     1303
26    1294
10    1294
19    1289
29    1286
4     1283
6     1281
11    1275
25    1274
7     1272
27    1258
5     1207
23    1206
30    1200
9     1171
31     849
Name: dia_ocu, dtype: int64


In [63]:
ht['fecha'] = ht['ano_ocu'].map(str) + '/' + ht['mes_ocu'].map(str) + '/' + ht['dia_ocu'].map(str) #concatenar la fecha completa en la columna fecha
ht['fecha'] = pd.to_datetime(ht['fecha'], dayfirst=False) #convertir los datos de la columna fecha en formato fecha
print(ht['fecha'].isnull().unique())

[False]


###**Fechas de vi**

In [64]:
vi.head(5)

ano_ocu  num_corre  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0     2015          1        1        1            4        16       3   
1     2015          2        1        1            4        16       3   
2     2015          3        1        1            4        22       4   
3     2015          4        1        1            4        22       4   
4     2015          5        1        1            4         2       1   

   g_hora_5  depto_ocu  mupio_ocu  ...  g_edad_80ymas  g_edad_60ymas  \
0         2          1        101  ...              3              3   
1         2          1        101  ...              2              2   
2         3          1        101  ...              3              3   
3         3          1        101  ...              5              5   
4         1          1        101  ...              2              2   

   edad_quinquenales  estado_con  tipo_veh  marca_veh  color_veh  modelo_veh  \
0                  5           9         1         32         11        2003   
1                  4           9         4         21          5        2011   
2                  5           1         1         21          3        9999   
3                  7           1         4         21          5        9999   
4                  4           9         4         21          1        2000   

   g_modelo_veh  tipo_eve  
0             4         1  
1             5         1  
2             6         1  
3             6         1  
4             4         1  

[5 rows x 24 columns]

In [65]:
print(vi['ano_ocu'].isnull().sum())
print(vi['mes_ocu'].isnull().sum())
print(vi['dia_ocu'].isnull().sum()) #no hay nulos en las fechas, se puede concatenar en una fecha completa en formato fecha
print(vi['ano_ocu'].value_counts())
print(vi['mes_ocu'].value_counts())
print(vi['dia_ocu'].value_counts()) #no se observan datos anormales en las fechas, meses o dias

0
0
0
2016    11618
2019    10827
2020    10103
2015     9823
2018     9514
2017     8644
Name: ano_ocu, dtype: int64
12    6040
3     5480
11    5348
2     5187
10    5022
1     5012
8     4961
9     4900
7     4839
4     4724
5     4590
6     4426
Name: mes_ocu, dtype: int64
1     2410
15    2169
2     2154
16    2139
18    2133
17    2083
12    2068
20    2041
22    2017
28    1999
13    1998
14    1990
21    1974
24    1963
3     1958
29    1953
8     1953
10    1922
26    1922
11    1912
27    1908
19    1908
7     1903
6     1895
4     1893
25    1884
30    1808
23    1805
5     1798
9     1739
31    1230
Name: dia_ocu, dtype: int64


In [66]:
vi['fecha'] = vi['ano_ocu'].map(str) + '/' + vi['mes_ocu'].map(str) + '/' + vi['dia_ocu'].map(str) #concatenar la fecha completa en la columna fecha
vi['fecha'] = pd.to_datetime(vi['fecha'], dayfirst=False) #convertir los datos de la columna fecha en formato fecha
print(vi['fecha'].isnull().unique())
vi.head(10)

[False]


ano_ocu  num_corre  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0     2015          1        1        1            4        16       3   
1     2015          2        1        1            4        16       3   
2     2015          3        1        1            4        22       4   
3     2015          4        1        1            4        22       4   
4     2015          5        1        1            4         2       1   
5     2015          6        1        1            4         2       1   
6     2015          7        1        1            4         9       2   
7     2015          8        1        1            4         1       1   
8     2015          9        1        1            4         8       2   
9     2015         10        1        1            4        17       3   

   g_hora_5  depto_ocu  mupio_ocu  ...  g_edad_60ymas  edad_quinquenales  \
0         2          1        101  ...              3                  5   
1         2          1        101  ...              2                  4   
2         3          1        101  ...              3                  5   
3         3          1        101  ...              5                  7   
4         1          1        101  ...              2                  4   
5         1          1        101  ...             12                 18   
6         1          1        101  ...              3                  5   
7         1          1        115  ...              3                  5   
8         1          1        114  ...              2                  4   
9         2          1        110  ...              3                  5   

   estado_con  tipo_veh  marca_veh  color_veh  modelo_veh  g_modelo_veh  \
0           9         1         32         11        2003             4   
1           9         4         21          5        2011             5   
2           1         1         21          3        9999             6   
3           1         4         21          5        9999             6   
4           9         4         21          1        2000             4   
5           9         3         44          6        9999             6   
6           9         4         40          5        9999             6   
7           9         1         34          4        9999             6   
8           1         4         41          4        9999             6   
9           1         7         22          7        9999             6   

   tipo_eve      fecha  
0         1 2015-01-01  
1         1 2015-01-01  
2         1 2015-01-01  
3         1 2015-01-01  
4         1 2015-01-01  
5         1 2015-01-01  
6         2 2015-01-01  
7         2 2015-01-01  
8         3 2015-01-01  
9         1 2015-01-01  

[10 rows x 25 columns]

In [67]:
vi.to_csv('/content/drive/MyDrive/vi.csv', index = False) # Guardamos el archivo como accidentes.csv


#**Código Preguntas**

## **1. ¿Cuál es el color de vehículo que más se accidenta? ¿Cómo se comporta ese color según la hora del día?**

In [ ]:
#@title ver columnas de vi
vi.columns

Index(['ano_ocu', 'num_corre', 'mes_ocu', 'dia_ocu', 'dia_sem_ocu', 'hora_ocu',
       'g_hora', 'g_hora_5', 'depto_ocu', 'mupio_ocu', 'zona_ocu', 'sexo_per',
       'edad_per', 'mayor_menor', 'g_edad_80ymas', 'g_edad_60ymas',
       'edad_quinquenales', 'estado_con', 'tipo_veh', 'marca_veh', 'color_veh',
       'modelo_veh', 'g_modelo_veh', 'tipo_eve', 'fecha'],
      dtype='object')

In [ ]:
#@title accidentes según el color del vehiculo
p1 = vi[['color_veh','hora_ocu']]
p1 = p1.groupby('color_veh')[['hora_ocu']].count().reset_index().rename(columns={'hora_ocu': 'cantidad'}).sort_values('cantidad', ascending=False)
p1.head(5)

color_veh  cantidad
16         99     15707
4           5      9619
0           1      8489
1           2      8182
3           4      6249

In [ ]:
#@title cambiar códigos por descripción
x1 = dicc.loc[dicc['Variable'] == 'color_veh'][['Codigo', 'Valor']].rename(columns={'Codigo': 'color_veh', 'Valor':'color'})
x1['color'] = x1['color'].replace(['Ignorado'], 'Sin dato')
p1 = pd.merge(p1, x1, how='left', on='color_veh')
p1

color_veh  cantidad           color
0          99     15707        Sin dato
1           5      9619           Negro
2           1      8489            Rojo
3           2      8182          Blanco
4           4      6249            Gris
5           3      4864            Azul
6           6      2242           Verde
7           9      1519         Corinto
8           7      1058        Amarillo
9          11      1037           Beige
10          8       486         Celeste
11         10       407            Café
12         14       394      Anaranjado
13         17       150  Varios colores
14         15        71          Morado
15         12        29        Turquesa
16         16        26          Rosado

In [ ]:
#@title Accidentes según el color del vehículo
fig1 = px.bar(p1, x='color', y='cantidad',
              title= '<b>Accidentes según el color del vehículo<b>')

fig1.update_layout(
    xaxis_title = 'Color del vehículo',
    yaxis_title = 'Número de accidentes',
    template = 'plotly_white',
    title_x = 0.5)

fig1.show()

In [ ]:
p11 = vi.loc[vi['color_veh'] == 5][['color_veh','hora_ocu']]
p11 = p11.groupby('hora_ocu')[['color_veh']].count().reset_index().rename(columns={'hora_ocu': 'hora', 'color_veh':'cantidad_negro'}).sort_values('hora')
p11 = p11.loc[p11['hora'] != 99]
p11

hora  cantidad_negro
0      0             460
1      1             308
2      2             249
3      3             196
4      4             153
5      5             160
6      6             232
7      7             355
8      8             344
9      9             310
10    10             354
11    11             338
12    12             360
13    13             343
14    14             416
15    15             458
16    16             443
17    17             581
18    18             646
19    19             630
20    20             624
21    21             623
22    22             547
23    23             487

In [ ]:
fig11 = px.bar(p11, x='hora', y='cantidad_negro',
               title= '<b>Accidentes de los vehículos negros según la hora del día<b>')

fig11.update_layout(
    xaxis_title = 'Hora del día',
    yaxis_title = 'Número de accidentes',
    template = 'plotly_white',
    title_x = 0.5)
fig11.show()

##**2. ¿Cómo es la evolución anual de los accidentes según el tipo de vehículo?**

In [ ]:
p2 = vi[['fecha', 'tipo_veh']]

x2 = dicc.loc[dicc['Variable'] == 'tipo_veh'][['Codigo', 'Valor']].rename(columns={'Codigo': 'tipo_veh', 'Valor':'tipo'})
p2 = pd.merge(p2, x2, how='left', on='tipo_veh')
p2 = p2.loc[p2['tipo_veh'] != 99]
p22 = p2.groupby(['tipo'])[['tipo_veh']].count().reset_index().rename(columns={'hora_ocu': 'hora', 'tipo_veh':'cantidad'}).sort_values('cantidad', ascending=False)
p22 = p22.head(5).reset_index().drop('index', axis=1)
p22["%"]= (p22["cantidad"]/p22["cantidad"].sum())*100
p22= p22.round(1)

tipo  cantidad
0               Motocicleta     22009
1                 Automóvil     12724
2                   Pick up      7639
3  Camioneta sport o blazer      3865
4                    Camión      3320

tipo  cantidad     %
0               Motocicleta     22009  44.4
1                 Automóvil     12724  25.7
2                   Pick up      7639  15.4
3  Camioneta sport o blazer      3865   7.8
4                    Camión      3320   6.7

In [ ]:
cnt_ed= p22["cantidad"].sum()

In [ ]:

fig = px.pie(p22, values = '%', names ='tipo',
             title= '<b>% de vehiculos involucrados en accidentes<b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

In [ ]:
fig2 = px.bar(p22, x='tipo', y='cantidad',
               title= '<b>Accidentes según el tipo de vehículo<b>')

fig2.update_layout(
    xaxis_title = 'Tipo de vehículo',
    yaxis_title = 'Número de accidentes',
    template = 'plotly_white',
    title_x = 0.5)
fig2.show()

In [ ]:
p23 = p2.loc[p2['tipo'].isin(list(p22['tipo'].unique()))].reset_index().drop('index', axis=1)
p23['fecha'] = p23['fecha'].dt.year
p23 = p23.groupby(['fecha','tipo'])[['tipo_veh']].count().reset_index().rename(columns={'tipo_veh':'cantidad'}).sort_values('fecha', ascending=True)
p23.head(5)

fecha                      tipo  cantidad
0   2015                 Automóvil      2172
1   2015  Camioneta sport o blazer       578
2   2015                    Camión       751
3   2015               Motocicleta      2755
4   2015                   Pick up      1470

In [ ]:
fig22 = px.line(p23, x='fecha', y='cantidad', 
                color='tipo',
                title = '<b>Evolución de los cinco vehículos con mayor accidentalidad<b>',)

fig22.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Tipo de vehículo:',
    xaxis_title = '<b>Año<b>',
    yaxis_title = '<b>Cantidad de accidentes<b>',
)

fig22.show()

##**3. ¿Cuáles son las ciudades con mayor accidentalidad de motocicletas? ¿Cómo ha sido la evolución anual?**


In [ ]:
p3 = vi[['fecha', 'mupio_ocu', 'tipo_veh']]

x3 = dicc.loc[dicc['Variable'] == 'tipo_veh'][['Codigo', 'Valor']].rename(columns={'Codigo': 'tipo_veh', 'Valor':'tipo_vehiculo'})
p3 = pd.merge(p3, x3, how='left', on='tipo_veh')
x3 = dicc.loc[dicc['Variable'] == 'mupio_ocu'][['Codigo', 'Valor']].rename(columns={'Codigo': 'mupio_ocu', 'Valor':'municipio'})
p3 = pd.merge(p3, x3, how='left', on='mupio_ocu')
p3 = p3.loc[p3['tipo_vehiculo'] == 'Motocicleta'][['fecha', 'tipo_vehiculo', 'municipio']]
p3

fecha tipo_vehiculo                  municipio
1     2015-01-01   Motocicleta                 Guatemala 
3     2015-01-01   Motocicleta                 Guatemala 
4     2015-01-01   Motocicleta                 Guatemala 
6     2015-01-01   Motocicleta                 Guatemala 
8     2015-01-01   Motocicleta                  Amatitlán
...          ...           ...                        ...
60518 2020-12-30   Motocicleta                     Panzós
60521 2020-12-30   Motocicleta                 San Benito
60526 2020-12-31   Motocicleta                      Ixcán
60527 2020-12-31   Motocicleta  Santa Lucía Cotzumalguapa
60528 2020-12-31   Motocicleta                  Escuintla

[22009 rows x 3 columns]

In [ ]:
pg3 = p3.groupby('municipio')[['tipo_vehiculo']].count().reset_index().rename(columns={'tipo_vehiculo':'cantidad'}).sort_values('cantidad', ascending=False)
pg3 = pg3.head(10).reset_index().drop('index', axis=1)
pg3['municipio'] = pg3['municipio'].replace({'Guatemala ': 'Ciudad de Guatemala', 'Petapa ': 'San Miguel Petapa'})
pg3

x3 = glc[['name','lat','lon']].rename(columns={'name':'municipio'})
pg3 = pd.merge(pg3, x3, how='left', on='municipio')
pg3 = pg3.drop(6).reset_index().drop('index', axis=1)
pg3

municipio  cantidad        lat        lon
0     Ciudad de Guatemala      5253  14.622233 -90.518519
1                   Mixco       829  14.630717 -90.606160
2               Escuintla       759  14.302286 -90.786624
3             Villa Nueva       684  14.530883 -90.596333
4       San Miguel Petapa       580  14.497691 -90.556033
5          Puerto Barrios       519  15.727515 -88.595251
6  San Pedro Sacatepéquez       373  14.686025 -90.642394
7           Chimaltenango       348  14.661304 -90.819349
8   San Juan Sacatepéquez       323  14.718432 -90.643406
9              Chiquimula       320  14.800571 -89.545917

In [ ]:
fig3 = px.bar(pg3, x='municipio', y='cantidad',
              title= '<b>Las 10 ciudades con mayor accidentalidad de motocicletas<b>'
)

fig3.update_layout(
    xaxis_title = 'Ciudad',
    yaxis_title = 'Número de accidentes',
    template = 'plotly_white',
    title_x = 0.5)

fig3.show()

In [ ]:
token_map = 'pk.eyJ1IjoiaW5nYWVtYXJpbiIsImEiOiJjbDBjeG94azUwMmZjM2tsY2MwNW81ejFoIn0.aAsgto5aYSLsBV1R3iOGug'
px.set_mapbox_access_token(token_map)

In [ ]:
fig33 = px.scatter_mapbox(pg3,
                          title='<b>10 Ciudades de Guatemala con mayor accidentalidad de motocicletas<b>', 
                          lat="lat", 
                          lon="lon", 
                          size="cantidad", 
                          color='cantidad',
                          color_continuous_scale='Viridis', 
                          center = {'lat':pg3['lat'].mean()+1, 'lon':pg3['lon'].mean()},
                          size_max=40, 
                          zoom=6.5,
                          height=800)

fig33.show()

In [ ]:
p33 = p3.copy()
p33['municipio'] = p33['municipio'].replace({'Guatemala ': 'Ciudad de Guatemala', 'Petapa ': 'San Miguel Petapa'}, value=None)
p33 = p33.loc[p33['municipio'].isin(list(pg3['municipio'].unique()))]
p33['fecha'] = p33['fecha'].dt.year
p33 = p33.groupby(['fecha','municipio'])[['tipo_vehiculo']].count().reset_index().rename(columns={'tipo_vehiculo':'cantidad'}).sort_values('fecha', ascending=True)
p33

fecha               municipio  cantidad
0    2015           Chimaltenango        46
1    2015              Chiquimula        45
2    2015     Ciudad de Guatemala       623
3    2015               Escuintla       118
4    2015                   Mixco        92
5    2015          Puerto Barrios        84
6    2015   San Juan Sacatepéquez        28
7    2015       San Miguel Petapa        16
8    2015  San Pedro Sacatepéquez         2
9    2015             Villa Nueva        89
19   2016             Villa Nueva         8
18   2016  San Pedro Sacatepéquez       130
17   2016       San Miguel Petapa       158
16   2016   San Juan Sacatepéquez        48
15   2016          Puerto Barrios       111
11   2016              Chiquimula        51
13   2016               Escuintla       156
12   2016     Ciudad de Guatemala       757
10   2016           Chimaltenango        78
14   2016                   Mixco        20
26   2017   San Juan Sacatepéquez        49
28   2017  San Pedro Sacatepéquez       122
27   2017       San Miguel Petapa       135
25   2017          Puerto Barrios        78
29   2017             Villa Nueva         4
23   2017               Escuintla       109
22   2017     Ciudad de Guatemala       780
21   2017              Chiquimula        39
20   2017           Chimaltenango        53
24   2017                   Mixco        18
39   2018             Villa Nueva        88
38   2018  San Pedro Sacatepéquez        95
37   2018       San Miguel Petapa       134
36   2018   San Juan Sacatepéquez        58
35   2018          Puerto Barrios        74
34   2018                   Mixco       113
33   2018               Escuintla       124
32   2018     Ciudad de Guatemala      1004
31   2018              Chiquimula        66
30   2018           Chimaltenango        45
48   2019  San Pedro Sacatepéquez        14
47   2019       San Miguel Petapa        56
49   2019             Villa Nueva       220
46   2019   San Juan Sacatepéquez        60
45   2019          Puerto Barrios        93
42   2019     Ciudad de Guatemala      1092
43   2019               Escuintla       119
41   2019              Chiquimula        58
40   2019           Chimaltenango        61
44   2019                   Mixco       278
58   2020  San Pedro Sacatepéquez        10
50   2020           Chimaltenango        65
51   2020              Chiquimula        61
52   2020     Ciudad de Guatemala       997
53   2020               Escuintla       133
54   2020                   Mixco       308
55   2020          Puerto Barrios        79
56   2020   San Juan Sacatepéquez        80
57   2020       San Miguel Petapa        81
59   2020             Villa Nueva       275

In [ ]:

#@title Evolución 10 ciudades
fig333 = px.line(p33, x='fecha', y='cantidad', 
                 color='municipio',
                 title = '<b>Evolución de las 10 ciudades con mayor accidentalidad de motocicletas<b>',)

fig333.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Municipio:',
    xaxis_title = '<b>Año<b>',
    yaxis_title = '<b>Cantidad de accidentes<b>',
)
fig333.show()

## **4. ¿Se presenta algún comportamiento trimestral en el tipo de accidente más recurrente? ¿Este cambia al presentarse anualmente?** ##

In [ ]:
#comportamiento trimestral
p4 = ht[['fecha', 'tipo_eve']]

x4 = dicc.loc[dicc['Variable'] == 'tipo_eve'][['Codigo', 'Valor']].rename(columns={'Codigo': 'tipo_eve', 'Valor':'tipo'})
p4 = pd.merge(p4, x4, how='left', on='tipo_eve')
p4 = p4.loc[p4['tipo_eve'] != 99]
p44 = p4.groupby(['tipo'])[['tipo_eve']].count().reset_index().rename(columns={'tipo_eve':'cantidad'}).sort_values('cantidad', ascending=False)
p44 = p44.head(5).reset_index().drop('index', axis=1)
p44["%"]= (p44["cantidad"]/p44["cantidad"].sum())*100
p44 = p44.round(1)


tipo  cantidad
0   Colisión     19631
1  Atropello     13178
2     Choque      3897
3     Vuelco      1369
4      Caída      1248

In [ ]:
p44["%"]= (p44["cantidad"]/p44["cantidad"].sum())*100
p44 = p44.round(1)
p44

tipo  cantidad     %
0   Colisión     19631  49.9
1  Atropello     13178  33.5
2     Choque      3897   9.9
3     Vuelco      1369   3.5
4      Caída      1248   3.2

In [ ]:
p42 = p4.loc[p4['tipo'].isin(list(p44['tipo'].unique()))]
p42 = p42.groupby(['fecha','tipo'])[['tipo_eve']].count().reset_index().rename(columns={'tipo_eve':'cantidad'}).sort_values('fecha', ascending=True)
p42

fecha      tipo  cantidad
0    2015-01-01  Colisión         6
1    2015-01-02  Colisión         6
2    2015-01-03  Colisión         9
3    2015-01-04  Colisión        10
4    2015-01-05  Colisión         5
...         ...       ...       ...
2178 2020-12-27  Colisión        14
2179 2020-12-28  Colisión        13
2180 2020-12-29  Colisión        13
2181 2020-12-30  Colisión        11
2182 2020-12-31  Colisión        19

[2183 rows x 3 columns]

In [ ]:
p42= p42.resample('Q',on ='fecha').sum().reset_index()
p42

fecha  cantidad
0  2015-03-31       668
1  2015-06-30       690
2  2015-09-30       782
3  2015-12-31       866
4  2016-03-31       898
5  2016-06-30       886
6  2016-09-30       902
7  2016-12-31       909
8  2017-03-31       628
9  2017-06-30       667
10 2017-09-30       690
11 2017-12-31       749
12 2018-03-31       843
13 2018-06-30       696
14 2018-09-30       708
15 2018-12-31       812
16 2019-03-31       979
17 2019-06-30       925
18 2019-09-30       875
19 2019-12-31       844
20 2020-03-31      1021
21 2020-06-30       655
22 2020-09-30       856
23 2020-12-31      1082

In [ ]:
#@title Evolución trimestral de colisiones en accidentes de transito
# definir gráfica
fig4 = px.line(p42, x='fecha', y = 'cantidad', title = '<b>Evolución trimestral de colisiones en accidentes de transito<b>',
              color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles
fig4.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    legend_title = 'Tipo de accidente:',
    xaxis_title = '<b>Fecha<b>',
    yaxis_title = '<b>Cantidad<b>',
)

fig4.show()

In [ ]:
p43= p42.resample('Y',on ='fecha').sum().reset_index()
p43

fecha  cantidad
0 2015-12-31      3006
1 2016-12-31      3595
2 2017-12-31      2734
3 2018-12-31      3059
4 2019-12-31      3623
5 2020-12-31      3614

In [ ]:
#@title Evolución anual de colisiones en accidentes de transito
fig41 = px.line(p43, x='fecha', y = 'cantidad', title = '<b>Evolución anual de colisiones en accidentes de transito<b>',
              color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles
fig41.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    legend_title = 'Tipo de accidente:',
    xaxis_title = '<b>Fecha<b>',
    yaxis_title = '<b>Cantidad<b>',
)

fig41.show()

## **5. ¿Como se distrubuyen las víctimas fallecidas y lesionadas por año?. ¿Cómo sería la distribución de fallecidos y lesionados según el género en cada caso?** ##

In [ ]:
fl.head(10)

num_corre  ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0          1     2015        1        1            4        16       3   
1          2     2015        1        1            4        16       3   
2          3     2015        1        1            4        22       4   
3          4     2015        1        1            4         2       1   
4          5     2015        1        1            4         9       2   
5          6     2015        1        1            4         1       1   
6          7     2015        1        1            4         8       2   
7          8     2015        1        1            4         8       2   
8          9     2015        1        1            4         8       2   
9         10     2015        1        1            4        17       3   

   g_hora_5  depto_ocu  mupio_ocu  ...  edad_quinquenales  fall_les  \
0         2          1        101  ...                  4         2   
1         2          1        101  ...                  4         2   
2         3          1        101  ...                  7         2   
3         1          1        101  ...                  4         2   
4         1          1        101  ...                  5         1   
5         1          1        115  ...                  5         1   
6         1          1        114  ...                  4         2   
7         1          1        114  ...                  4         2   
8         1          1        114  ...                 11         2   
9         2          1        110  ...                  5         2   

   int_o_noint  tipo_veh  marca_veh  color_veh  modelo_veh  g_modelo_veh  \
0            1         4         21          5        2011             5   
1            1         4         21          5        2011             5   
2            1         4         21          5        9999             6   
3            1         4         21          1        2000             4   
4            2         4         40          5        9999             6   
5            2         1         34          4        9999             6   
6            1         4         41          4        9999             6   
7            1         4         41          4        9999             6   
8            1         4         41          4        9999             6   
9            1         4         21          1        9999             6   

   tipo_eve      fecha  
0         1 2015-01-01  
1         1 2015-01-01  
2         1 2015-01-01  
3         1 2015-01-01  
4         2 2015-01-01  
5         2 2015-01-01  
6         3 2015-01-01  
7         3 2015-01-01  
8         3 2015-01-01  
9         1 2015-01-01  

[10 rows x 26 columns]

In [ ]:
    p5 = fl[['ano_ocu', 'fall_les','sexo_per']]
    x5 = dicc.loc[dicc['Variable'] == 'fall_les'][['Codigo', 'Valor']].rename(columns={'Codigo': 'fall_les', 'Valor':'tipo'})
    p5 = pd.merge(p5, x5, how='left', on='fall_les')
    x53 = dicc.loc[dicc['Variable'] == 'sexo_per'][['Codigo', 'Valor']].rename(columns={'Codigo': 'sexo_per', 'Valor':'tipo'})
    p53 = pd.merge(p5, x53, how='left', on='sexo_per')
    p53 = p53.rename(columns={'tipo_x': 'tipo', 'tipo_y':'sexo'})
    p57lh= p53[p53["tipo"]== 'Lesionado']
    p58lh= p57lh[p57lh["sexo"]== 'Hombre']
    p58lh = p58lh.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
    p57lm= p53[p53["tipo"]== 'Lesionado']
    p58lm= p57lm[p57lm["sexo"]== 'Mujer']
    p58lm = p58lm.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
    xy= pd.merge(p58lh,p58lm, how= "outer",on='ano_ocu').reset_index().rename(columns={"Cantidad_x":"Hombre",'Cantidad_y':'Mujer'})
    xy

ano_ocu  Hombre  Mujer
0     2015    5866   2828
1     2016    6464   3181
2     2017    4962   2175
3     2018    5446   2486
4     2019    5997   2861
5     2020    4847   1716

In [ ]:
xy["%h"]= (xy["Hombre"]/xy["Hombre"].sum())*100
xy= xy.round(1)
xy["%m"]= (xy["Mujer"]/xy["Mujer"].sum())*100
xy= xy.round(1)
xy

ano_ocu  Hombre  Mujer    %h    %m
0     2015    5866   2828  17.5  18.5
1     2016    6464   3181  19.2  20.9
2     2017    4962   2175  14.8  14.3
3     2018    5446   2486  16.2  16.3
4     2019    5997   2861  17.9  18.8
5     2020    4847   1716  14.4  11.3

In [ ]:
cnt_ed = xy['Hombre'].sum()

# hacer la gráfica
fig = px.pie(xy, values = '%h', names ='ano_ocu',
             title= '<b>% Edificios según la calidad de la calefacción<b>', hole = .5,
             color_discrete_sequence=px.colors.qualitative.G10)

# poner detalles a la gráfica
fig.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    annotations = [dict(text = str(cnt_ed), x=0.5, y = 0.5, font_size = 40, showarrow = False )])
fig.show()

In [ ]:
fig = px.pie(xy, values = '%h', names ='ano_ocu',
             title= '<b>Edificios vendidos según la calidad del material en el exterior<b>',
             color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles a la gráfica:
fig.update_layout(
    xaxis_title = 'Año',
    yaxis_title = 'Precio de venta',
    template = 'simple_white',
    title_x = 0.5)

fig.show()

In [ ]:
p57fh= p53[p53["tipo"]== 'Fallecido']
p58fh= p57fh[p57fh["sexo"]== 'Hombre']
p58fh = p58fh.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
p57fm= p53[p53["tipo"]== 'Fallecido']
p58fm= p57fm[p57fm["sexo"]== 'Mujer']
p58fm = p58fm.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
xy1= pd.merge(p58fh,p58fm, how= "left",on='ano_ocu').reset_index().rename(columns={"Cantidad_x":"Hombre",'Cantidad_y':'Mujer'})
xy1

ano_ocu  Hombre  Mujer
0     2015    1403    261
1     2016    1676    324
2     2017    1218    256
3     2018    1221    244
4     2019    1433    306
5     2020    1314    221

In [ ]:
xy1["%h"]= (xy1["Hombre"]/xy1["Hombre"].sum())*100
xy1= xy1.round(1)
xy1["%m"]= (xy1["Mujer"]/xy1["Mujer"].sum())*100
xy1= xy1.round(1)
xy1

ano_ocu  Hombre  Mujer    %h    %m
0     2015    1403    261  17.0  16.2
1     2016    1676    324  20.3  20.1
2     2017    1218    256  14.7  15.9
3     2018    1221    244  14.8  15.1
4     2019    1433    306  17.3  19.0
5     2020    1314    221  15.9  13.7

In [ ]:
p5 = fl[['ano_ocu', 'fall_les',]]
x5 = dicc.loc[dicc['Variable'] == 'fall_les'][['Codigo', 'Valor']].rename(columns={'Codigo': 'fall_les', 'Valor':'tipo'})
p5 = pd.merge(p5, x5, how='left', on='fall_les')
p5= p5.groupby(['ano_ocu', 'tipo'])[['fall_les']].count().reset_index().rename(columns={'fall_les':'cantidad'}) # agrupar por dos columnas
p5

ano_ocu       tipo  cantidad
0      2015  Fallecido      1679
1      2015  Lesionado      8718
2      2016  Fallecido      2008
3      2016  Lesionado      9660
4      2017  Fallecido      1480
5      2017  Lesionado      7145
6      2018  Fallecido      1469
7      2018  Lesionado      7938
8      2019  Fallecido      1757
9      2019  Lesionado      8903
10     2020  Fallecido      1549
11     2020  Lesionado      6593

In [ ]:
#@title Evolución anual de fallecidos y lesionados accidentes de transito
fig51 = px.bar(p5, x='ano_ocu', y = 'cantidad',barmode= 'group',color = 'tipo' ,title = '<b>Evolución anual de fallecidos y lesionados accidentes de transito<b>',
              color_discrete_sequence=px.colors.qualitative.G10)

# agregar detalles
fig51.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    legend_title = 'Tipo de accidente:',
    xaxis_title = '<b>Fecha<b>',
    yaxis_title = '<b>Cantidad<b>',
)

fig51.show()

**Analisis de gráfica:** De acuerdo a la gráfica,se observa una gran diferencia en cuanto a los lesionados y fallecidos. Donde el mayor año donde se presentaron mayores atentidos en centros médicos fue el 2016 esto se debe a que no se establecieron estrictas normativas en todo el país por lo que se tenia cierta libertad para la movilidad. 

Por cada tipo de evento(fallecido, lesionados) y género:

In [ ]:

p5 = fl[['ano_ocu', 'fall_les','sexo_per']]
x5 = dicc.loc[dicc['Variable'] == 'fall_les'][['Codigo', 'Valor']].rename(columns={'Codigo': 'fall_les', 'Valor':'tipo'})
p5 = pd.merge(p5, x5, how='left', on='fall_les')
x53 = dicc.loc[dicc['Variable'] == 'sexo_per'][['Codigo', 'Valor']].rename(columns={'Codigo': 'sexo_per', 'Valor':'tipo'})
p53 = pd.merge(p5, x53, how='left', on='sexo_per')
p53 = p53.rename(columns={'tipo_x': 'tipo', 'tipo_y':'sexo'})
p53

ano_ocu  fall_les  sexo_per       tipo    sexo
0         2015         2         1  Lesionado  Hombre
1         2015         2         2  Lesionado   Mujer
2         2015         2         1  Lesionado  Hombre
3         2015         2         1  Lesionado  Hombre
4         2015         1         1  Fallecido  Hombre
...        ...       ...       ...        ...     ...
58898     2020         2         2  Lesionado   Mujer
58899     2020         2         2  Lesionado   Mujer
58900     2020         2         1  Lesionado  Hombre
58901     2020         1         2  Fallecido   Mujer
58902     2020         2         2  Lesionado   Mujer

[58903 rows x 5 columns]

In [ ]:
p57lh= p53[p53["tipo"]== 'Lesionado']
p58lh= p57lh[p57lh["sexo"]== 'Hombre']
p58lh = p58lh.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
p57lm= p53[p53["tipo"]== 'Lesionado']
p58lm= p57lm[p57lm["sexo"]== 'Mujer']
p58lm = p58lm.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
xy= pd.merge(p58lh,p58lm, how= "outer",on='ano_ocu').reset_index().rename(columns={"Cantidad_x":"Hombre",'Cantidad_y':'Mujer'})
xy

ano_ocu  Hombre  Mujer
0     2015    5866   2828
1     2016    6464   3181
2     2017    4962   2175
3     2018    5446   2486
4     2019    5997   2861
5     2020    4847   1716

In [ ]:
#@title Lesionados de acuerdo al tipo de sexo
fig5 = px.bar(xy, x = 'ano_ocu', y=['Hombre','Mujer'], barmode= 'group', title= '<b>Lesionados de acuerdo al tipo de sexo<b>')

# agregar detalles a la gráfica
fig5.update_layout(
    xaxis_title = 'Año',
    yaxis_title = 'Número de Accidentes',
    template = 'simple_white',
    title_x = 0.5)

fig5.show()

**Análisis de gráfica:** A través de la gráfica se ve reflejado que los hombres son los que más se accidentan y salen lesionados. Se puede apreciar que en el 2017 estos accidentes redujeron con un total de 7.137 teniendo en cuanto ambos generos.
Mientras que en el año 2016 sigue teniendo el comportamiento de anteriores gráficas, es decir, el año con más hechos. En este caso con un total de 6464 para hombres y 3181 para mujeres.

In [ ]:
p57fh= p53[p53["tipo"]== 'Fallecido']
p58fh= p57fh[p57fh["sexo"]== 'Hombre']
p58fh = p58fh.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
p57fm= p53[p53["tipo"]== 'Fallecido']
p58fm= p57fm[p57fm["sexo"]== 'Mujer']
p58fm = p58fm.groupby(['ano_ocu','sexo']).agg({'tipo':'count'}).rename(columns={'tipo':'Cantidad'}) # no aplicarmos reset
xy1= pd.merge(p58fh,p58fm, how= "left",on='ano_ocu').reset_index().rename(columns={"Cantidad_x":"Hombre",'Cantidad_y':'Mujer'})
xy1

ano_ocu  Hombre  Mujer
0     2015    1403    261
1     2016    1676    324
2     2017    1218    256
3     2018    1221    244
4     2019    1433    306
5     2020    1314    221

In [ ]:
#@title Fallecidos de acuerdo al tipo de sexo
fig52 = px.bar(xy1, x = 'ano_ocu', y=['Hombre','Mujer'], barmode= 'group', title= '<b>Fallecidos de acuerdo al tipo de sexo<b>')

# agregar detalles a la gráfica
fig52.update_layout(
    xaxis_title = 'Año',
    yaxis_title = 'Número de Accidentes',
    template = 'simple_white',
    title_x = 0.5)

fig52.show()

## **6. Por día de la semana, ¿cómo se distribuyen los accidentes de tránsito? ¿Cómo se distribuyen los fallecidos y lesionados?**

In [ ]:
fl6 = fl.copy()
fl6['dia_sem'] = fl6['fecha'].dt.day_name()
fl6[58650:58710]
fl6

num_corre  ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu  hora_ocu  g_hora  \
0             1     2015        1        1            4        16       3   
1             2     2015        1        1            4        16       3   
2             3     2015        1        1            4        22       4   
3             4     2015        1        1            4         2       1   
4             5     2015        1        1            4         9       2   
...         ...      ...      ...      ...          ...       ...     ...   
8137       8138     2020       12       31            4        13       3   
8138       8139     2020       12       31            4        13       3   
8139       8140     2020       12       31            4        13       3   
8140       8141     2020       12       31            4        18       4   
8141       8142     2020       12       31            4        21       4   

      g_hora_5  depto_ocu  mupio_ocu  ...  fall_les  int_o_noint  tipo_veh  \
0            2          1        101  ...         2            1         4   
1            2          1        101  ...         2            1         4   
2            3          1        101  ...         2            1         4   
3            1          1        101  ...         2            1         4   
4            1          1        101  ...         1            2         4   
...        ...        ...        ...  ...       ...          ...       ...   
8137         2         14       1420  ...         2            1         4   
8138         2         14       1420  ...         2            1         4   
8139         2         14       1420  ...         2            1         4   
8140         3          5        502  ...         1            2         4   
8141         3          5        501  ...         2            1         4   

      marca_veh  color_veh  modelo_veh  g_modelo_veh  tipo_eve      fecha  \
0            21          5        2011             5         1 2015-01-01   
1            21          5        2011             5         1 2015-01-01   
2            21          5        9999             6         1 2015-01-01   
3            21          1        2000             4         1 2015-01-01   
4            40          5        9999             6         2 2015-01-01   
...         ...        ...         ...           ...       ...        ...   
8137          2          5        9999            99         5 2020-12-31   
8138          2          5        9999            99         5 2020-12-31   
8139          2          5        9999            99         5 2020-12-31   
8140        168          1        2019             5         5 2020-12-31   
8141         27          1        2013             5         5 2020-12-31   

       dia_sem  
0     Thursday  
1     Thursday  
2     Thursday  
3     Thursday  
4     Thursday  
...        ...  
8137  Thursday  
8138  Thursday  
8139  Thursday  
8140  Thursday  
8141  Thursday  

[58903 rows x 27 columns]

In [ ]:
#SE QUIERE COMPROBAR SI EL DIA DE LA SEMANA SUMINISTRADO POR LA BASE DE DATOS COINCIDE 
#CON LOS VALORES OBTENIDOS AL USAR LA FUNCION .dt.day_name()

#SE EXTRAEN LOS VALORES DE LOS NOMBRES DE LOS DIAS DE LA SEMANA CON SU INDICADOR O CODIGO. EJE: 1 = Lunes, 2 = Martes, ...
x6 = dicc.loc[dicc['Variable'] == 'día_sem_ocu'][['Codigo', 'Valor']].rename(columns={'Codigo': 'dia_sem_ocu', 'Valor':'nombre_dia'})
x6['nombre_dia'] = x6['nombre_dia'].replace(['Ignorado'], 'Sin dato')
#x6


#Se agrega la columena nombre_dia con el merge, el cual es el nombre del día según el codigo de la columna dia_sem_ocu
fl6 = pd.merge(fl6, x6, how='left', on='dia_sem_ocu')
fl6

#Se imprimen los nombres de los días con su cantidad, obtenidos con to_dayname()
print(fl6['dia_sem'].value_counts())

#Se imprimen los nombres de los días con su cantidad, obtenidos a partir de la columna dia_sem_ocu
print(fl6['nombre_dia'].value_counts())

#No se obtienen la misma cantidad en cada día. Se debe verificar cual columna está arrojando valores incorrectos

Sunday       11294
Saturday     10759
Friday        8100
Monday        7960
Thursday      7366
Tuesday       6841
Wednesday     6583
Name: dia_sem, dtype: int64
Domingo      11298
Sábado       10781
Viernes       8109
Lunes         7946
Jueves        7365
Martes        6816
Miércoles     6588
Name: nombre_dia, dtype: int64


![](https://media.istockphoto.com/vectors/calendar-2019-2020-illustration-week-starts-on-sunday-vector-id1016807090)

In [ ]:
#Se crea una bd temporal donde se pueda comparar cada columna
#dia_sem es el atributo del nombre del dia de la semana obtenido a partir de la funcion .dt.day_name()
#nombre_dia es el atributo del nombre del dia de la semana obtenido a partir del atributo 'dia_sem_ocu' suministrado por la bd
temp = fl6.loc[fl6['nombre_dia'] == 'Sábado'][['ano_ocu','mes_ocu','dia_ocu', 'dia_sem_ocu', 'fecha','dia_sem','nombre_dia']]
temp['nombre_dia'] = temp['nombre_dia'].replace('Sábado', 'Saturday')
temp.loc[temp['dia_sem'] != 'Saturday'][30:]

#Comparando con el calendario de 2019 y 2020, vemos que el 
#día de la semana coincide con los valores del atributo dia_sem, obtenidos a partir de fl['fecha'].dt.day_name()

ano_ocu  mes_ocu  dia_ocu  dia_sem_ocu      fecha dia_sem nombre_dia
46234     2019        7       28            6 2019-07-28  Sunday   Saturday
46235     2019        7       28            6 2019-07-28  Sunday   Saturday
55001     2020        7       10            6 2020-07-10  Friday   Saturday

In [ ]:
p6 = fl.copy()
p6['dia_sem'] = p6['fecha'].dt.day_name()
p6['dia_sem'] = p6['dia_sem'].replace({'Monday': 'Lunes', 
                                       'Tuesday': 'Martes',
                                       'Wednesday': 'Miércoles',
                                       'Thursday': 'Jueves',
                                       'Friday': 'Viernes',
                                       'Saturday': 'Sábado',
                                       'Sunday': 'Domingo'})
p6['dia_sem'].unique()

array(['Jueves', 'Viernes', 'Sábado', 'Domingo', 'Lunes', 'Martes',
       'Miércoles'], dtype=object)

In [ ]:
pg6  = p6.groupby('dia_sem')[['fecha']].count().reset_index().rename(columns={'fecha':'cantidad_eventos'})
pg6['num_dia_semana'] = pg6['dia_sem'].apply(num_dia_sem)
pg6 = pg6[['num_dia_semana','dia_sem','cantidad_eventos']].sort_values('num_dia_semana', ascending=True)
pg6['proporcion'] = pg6['cantidad_eventos'].apply(lambda x: round(x/pg6['cantidad_eventos'].sum()*100,2))
pg6

num_dia_semana    dia_sem  cantidad_eventos  proporcion
2               1      Lunes              7960       13.51
3               2     Martes              6841       11.61
4               3  Miércoles              6583       11.18
1               4     Jueves              7366       12.51
6               5    Viernes              8100       13.75
5               6     Sábado             10759       18.27
0               7    Domingo             11294       19.17

In [ ]:
fig6 = px.bar(pg6, x='dia_sem', y='cantidad_eventos',
              title = '<b>Accidentalidad según el día<b>',)

fig6.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    xaxis_title = '<b>Día de la semana<b>',
    yaxis_title = '<b>Cantidad de accidentes<b>',
)
fig6.show()

In [ ]:
#Se extraen los atributos de interes
p66 = p6[['dia_sem', 'fall_les','fecha']]

#Se cambiaran los valores de 1, 0 o 9 de fall_les por su descripción: fallecido, lesionado, Sin dato. Los cuales están en el dicc
x66 = dicc.loc[dicc['Variable'] == 'fall_les'][['Codigo', 'Valor']].rename(columns={'Codigo': 'fall_les', 'Valor':'fallecido_lesionado'})
p66 = pd.merge(p66, x66, how='left', on='fall_les')
#No se van a eliminar los registros del atributo fall_les donde hay 9 ya que se puede obtener información de estos registros.
#Se cambiaran por Sin Dato

p66['fallecido_lesionado'] = p66['fallecido_lesionado'].fillna('Sin dato')
p66

pg66 = p66.groupby(['dia_sem','fallecido_lesionado'])[['fecha']].count().reset_index().rename(columns={'fecha':'cantidad'})
pg66['num_dia_semana'] = pg66['dia_sem'].apply(num_dia_sem)
pg66 = pg66[['num_dia_semana', 'dia_sem','fallecido_lesionado','cantidad']].sort_values('num_dia_semana')
pg66

num_dia_semana    dia_sem fallecido_lesionado  cantidad
5                1      Lunes           Fallecido      1295
6                1      Lunes           Lesionado      6665
7                2     Martes           Fallecido      1045
8                2     Martes           Lesionado      5794
9                2     Martes            Sin dato         2
10               3  Miércoles           Fallecido      1071
11               3  Miércoles           Lesionado      5512
3                4     Jueves           Fallecido      1214
4                4     Jueves           Lesionado      6152
14               5    Viernes           Fallecido      1349
15               5    Viernes           Lesionado      6751
12               6     Sábado           Fallecido      1910
13               6     Sábado           Lesionado      8849
0                7    Domingo           Fallecido      2058
1                7    Domingo           Lesionado      9234
2                7    Domingo            Sin dato         2

In [ ]:
fig66 = px.bar(pg66, x='dia_sem', y='cantidad', 
               color = 'fallecido_lesionado', barmode= 'group',
               title = '<b>Gravedad según el día de la semana<b>',)

fig66.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Gravedad:',
    xaxis_title = '<b>Día de la semana<b>',
    yaxis_title = '<b>Cantidad de accidentes<b>',
)
fig66.show()

## **7. ¿Cómo es la proporción de internados en centros médicos? ¿Cómo cambia según el género? ¿Cómo afecta el tipo de evento la cantidad de internados?**

In [ ]:
p7 = fl[['fecha', 'int_o_noint', 'sexo_per', 'tipo_eve']]
p7

x7 = dicc.loc[dicc['Variable'] == 'int_o_noint'][['Codigo', 'Valor']].rename(columns={'Codigo': 'int_o_noint', 'Valor':'estado'})
x7['estado'] = x7['estado'].replace('Ignorado','Sin dato')
p7 = pd.merge(p7, x7, how='left', on='int_o_noint')

x7 = dicc.loc[dicc['Variable'] == 'sexo_per'][['Codigo', 'Valor']].rename(columns={'Codigo': 'sexo_per', 'Valor':'sexo'})
x7['sexo'] = x7['sexo'].replace('Ignorado','Sin dato')
p7 = pd.merge(p7, x7, how='left', on='sexo_per')

x7 = dicc.loc[dicc['Variable'] == 'tipo_eve'][['Codigo', 'Valor']].rename(columns={'Codigo': 'tipo_eve', 'Valor':'evento'})
x7['evento'] = x7['evento'].replace('Ignorado','Sin dato')
p7 = pd.merge(p7, x7, how='left', on='tipo_eve')
p7 = p7.drop(['int_o_noint', 'sexo_per', 'tipo_eve'], axis=1)

In [ ]:
pgi7 = p7.groupby('estado')[['fecha']].count().reset_index().rename(columns={'fecha':'cantidad'})
display(pgi7)

estado  cantidad
0     Internado     45442
1  No internado     13202
2      Sin dato       259

In [ ]:
fig7 = px.pie(pgi7, values='cantidad', names='estado',
              title = '<b>Proporción de internados<b>')

fig7.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Estado:',
    height=700,
    width=700
)

fig7.show()

In [ ]:
pgg7 = p7.groupby(['estado', 'sexo'])[['fecha']].count().reset_index().rename(columns={'fecha':'cantidad'})
display(pgg7)

estado      sexo  cantidad
0     Internado    Hombre     31375
1     Internado     Mujer     13957
2     Internado  Sin dato       110
3  No internado    Hombre     10306
4  No internado     Mujer      2817
5  No internado  Sin dato        79
6      Sin dato    Hombre       170
7      Sin dato     Mujer        85
8      Sin dato  Sin dato         4

In [ ]:
fig77 = px.bar(pgg7, x='estado', y='cantidad', 
               color = 'sexo', barmode= 'group',
               title = '<b>Distribución del género según el estado<b>')

fig77.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Género:',
    xaxis_title = '<b>Estado<b>',
    yaxis_title = '<b>Cantidad<b>',
    height=700
)

fig77.show()

In [ ]:
pge7 = p7.groupby(['estado', 'evento'])[['fecha']].count().reset_index().rename(columns={'fecha':'cantidad'})
display(pge7)

estado      evento  cantidad
0      Internado   Atropello     10870
1      Internado       Caída      1876
2      Internado      Choque      4342
3      Internado    Colisión     23244
4      Internado     Derrape       836
5      Internado  Embarranco       255
6      Internado    Encuneto         7
7      Internado    Sin dato        10
8      Internado      Vuelco      4002
9   No internado   Atropello      4090
10  No internado       Caída       745
11  No internado      Choque      1637
12  No internado    Colisión      5087
13  No internado     Derrape       399
14  No internado  Embarranco       117
15  No internado    Encuneto         2
16  No internado    Sin dato         3
17  No internado      Vuelco      1122
18      Sin dato   Atropello        47
19      Sin dato       Caída        75
20      Sin dato      Choque        27
21      Sin dato    Colisión        97
22      Sin dato      Vuelco        13

In [ ]:
fig777 = px.bar(pge7, x='evento', y='cantidad', color = 'estado', barmode= 'group',
               title = '<b>Distribución de los internados según el tipo de accidente<b>')

fig777.update_layout(
    template = 'plotly_white',
    title_x = 0.5,
    legend_title = 'Estado:',
    xaxis_title = '<b>Tipo de accidente<b>',
    yaxis_title = '<b>Cantidad<b>',
    height=500
)

fig777.show()

## **8. ¿ Cantidad de personas involucradas en eventos de transito en estado de embriaguez por mes y por departamento?**

In [ ]:
vi.columns

Index(['ano_ocu', 'num_corre', 'mes_ocu', 'dia_ocu', 'dia_sem_ocu', 'hora_ocu',
       'g_hora', 'g_hora_5', 'depto_ocu', 'mupio_ocu', 'zona_ocu', 'sexo_per',
       'edad_per', 'mayor_menor', 'g_edad_80ymas', 'g_edad_60ymas',
       'edad_quinquenales', 'estado_con', 'tipo_veh', 'marca_veh', 'color_veh',
       'modelo_veh', 'g_modelo_veh', 'tipo_eve', 'fecha'],
      dtype='object')

In [ ]:
p8= vi[['mes_ocu','depto_ocu','estado_con']]
p8

mes_ocu  depto_ocu  estado_con
0            1          1           9
1            1          1           9
2            1          1           1
3            1          1           1
4            1          1           9
...        ...        ...         ...
10098       12         16           9
10099       12          2           9
10100       12         14           9
10101       12          5           1
10102       12          5           2

[60529 rows x 3 columns]

In [ ]:
x8 = dicc.loc[dicc['Variable'] == 'depto_ocu'][['Codigo', 'Valor']].rename(columns={'Codigo': 'depto_ocu', 'Valor':'tipo'})
x88 = dicc.loc[dicc['Variable'] == 'estado_con'][['Codigo', 'Valor']].rename(columns={'Codigo': 'estado_con', 'Valor':'tipo'})


In [ ]:
p8= pd.merge(p8, x8, how='left', on='depto_ocu')
p8=pd.merge(p8,x88, how='left', on='estado_con').rename(columns={'tipo_x':'depto','tipo_y':'estado'}).drop(['depto_ocu', 'estado_con'], axis=1)
p8

mes_ocu         depto    estado
0            1     Guatemala  Ignorado
1            1     Guatemala  Ignorado
2            1     Guatemala  No ebrio
3            1     Guatemala  No ebrio
4            1     Guatemala  Ignorado
...        ...           ...       ...
60524       12  Alta Verapaz  Ignorado
60525       12   El Progreso  Ignorado
60526       12        Quiché  Ignorado
60527       12     Escuintla  No ebrio
60528       12     Escuintla     Ebrio

[60529 rows x 3 columns]

In [ ]:
p88= p8[p8['estado']=='Ebrio'].groupby(['mes_ocu','depto'])[['estado']].count().reset_index().rename(columns={'estado':'cantidad_ebrios'})
p88

mes_ocu          depto  cantidad_ebrios
0          1   Alta Verapaz               28
1          1   Baja Verapaz               18
2          1  Chimaltenango               14
3          1     Chiquimula               11
4          1    El Progreso                8
..       ...            ...              ...
258       12     Santa Rosa               30
259       12         Sololá                7
260       12  Suchitepéquez               19
261       12    Totonicapán                3
262       12         Zacapa               17

[263 rows x 3 columns]

In [ ]:
# crear dataset
#base = d.groupby(['MSSubClass', 'YrSold'])[['SalePrice']].sum().reset_index()
# MSSubClass: Tipo de edificio
# YrSold: Año en que se vende
# SalePrice: Precio de venta

# crear gráfica
fig8 = px.bar(p88, x = 'mes_ocu', y='cantidad_ebrios', color = 'depto', 
             title= '<b>Cantidad de personas involucradas en eventos de transito en estado de embriaguez por mes y por ciudad<b>')

# agregar detalles a la gráfica
fig8.update_layout(
    xaxis_title = 'Mes',
    yaxis_title = 'Cantidad',
    template = 'simple_white',
    title_x = 0.5)

fig8.show()



## **9.¿Cómo se comporta el número de fallecidos,  ebrios y colisiones según el mes? Serie diaria**

In [ ]:
fl9=fl[fl['fall_les']==1].groupby(['mes_ocu'])[['fall_les']].count().reset_index().rename(columns={'fall_les': 'cantidad fallecidos'})
vi9= vi[vi['estado_con']==2].groupby(['mes_ocu'])[['estado_con']].count().reset_index().rename(columns={'estado_con': 'cantidad_ebrios'})
ht9=ht[ht['tipo_eve']==1].groupby(['mes_ocu'])[['tipo_eve']].count().reset_index().rename(columns={'tipo_eve': 'cantidad_colisiones'})
display(fl9,vi9,ht9)

mes_ocu  cantidad fallecidos
0         1                  807
1         2                  848
2         3                  988
3         4                  752
4         5                  726
5         6                  680
6         7                  801
7         8                  777
8         9                  813
9        10                  795
10       11                  841
11       12                 1114

mes_ocu  cantidad_ebrios
0         1              430
1         2              438
2         3              469
3         4              448
4         5              412
5         6              401
6         7              484
7         8              468
8         9              515
9        10              417
10       11              472
11       12              534

mes_ocu  cantidad_colisiones
0         1                 1617
1         2                 1651
2         3                 1769
3         4                 1516
4         5                 1562
5         6                 1441
6         7                 1571
7         8                 1631
8         9                 1611
9        10                 1634
10       11                 1735
11       12                 1893

In [ ]:
p9=pd.merge(fl9, vi9, how='outer', on='mes_ocu')
p9=pd.merge(p9, ht9, how='outer', on='mes_ocu')
p9

mes_ocu  cantidad fallecidos  cantidad_ebrios  cantidad_colisiones
0         1                  807              430                 1617
1         2                  848              438                 1651
2         3                  988              469                 1769
3         4                  752              448                 1516
4         5                  726              412                 1562
5         6                  680              401                 1441
6         7                  801              484                 1571
7         8                  777              468                 1631
8         9                  813              515                 1611
9        10                  795              417                 1634
10       11                  841              472                 1735
11       12                 1114              534                 1893

In [ ]:
columnas = list(p9.columns)
columnas = columnas[1:]
columnas

fig9 = px.line(p9, x='mes_ocu', y = columnas , 
              color_discrete_sequence=px.colors.qualitative.G10,
               title='<b>Evolución por mes de fallecidos, ebrio y colisiones<b>')

fig9.update_layout(
    template = 'simple_white',
    title_x = 0.5,
    legend_title = 'Tipo:',
    xaxis_title = '<b>Mes<b>',
    yaxis_title = '<b>Cantidad<b>',
)

fig9.show()

#**RESUMEN**

##**1. ¿Cuál es el color de vehículo que más se accidenta? ¿Cómo se comporta ese color según la hora del día?**

In [ ]:
fig1.show()
fig11.show()

**Análisis**

El color de vehículo con mayor accidentalidad es el negro seguido por el rojo. Esto se puede deber a que son colores que en la noche no son tan visibles como otros colores. En la figura de accidentes de los vehículos negros según la hora, se observa que la mayor accidentalidad ocurre en horarios nocturnos.

Cabe aclarar que los vehículos de color negro, rojo y blanco podrían ser los más vendidos, por lo tanto aumentarían la cantidad de siniestros de este tipo al tener una mayor población.

Por otro lado, existe una gran cantidad de colores sin información, no se conoce la razón por la cual no se registró el color del vehículo.

##**2. ¿Cómo es la evolución anual de los accidentes según el tipo de vehículo?**

In [ ]:
fig2.show()
fig22.show()

**Análisis**

La mayor cantidad de accidentes se presenta en motocicletas seguido por los automóviles. Se podría tener en cuenta que también son lo vehículos de mayor influencia vehicular.

Las motocicletas han sido en todos los años las de mayor accidentalidad, y se puede observar que apartir del año 2017 su cantidad de accidentes ha ido en aumento.

## **3. ¿Cuáles son las ciudades con mayor accidentalidad de motocicletas? ¿Cómo ha sido la evolución anual?**

In [ ]:
fig3.show()
fig33.show()
fig333.show()

**Análisis**

La mayor cantidad de accidentes de motocicletas se presenta en la ciudad de Guatemala, el cual es cinco veces mayor que las otras 10 ciudades donde más se presenta el accidente de este vehículo.

Durante los años analizados, en las ciudades de Mixco y Villanueva los accidentes de motocicletas se han disparado de 2017 a 2020. En la ciudad de Guatemala este comoportamiento era similar, pero luego del 2019 tendió a disminuir.

##**4. ¿Se presenta algún comportamiento trimestral en el tipo de accidente más recurrente? ¿Este cambia al presentarse anualmente?**

In [ ]:
fig4.show()
fig41.show()

**Analisis**


De acuerdo a las gráficas planteadas, se nota un mejor comportamiento trimestral,en el cual existe un seguimiento más especificado de los fallecidos en accidentes de transito. Se observa también, que en los primeros trimestres se presenta un aumento donde se registra el primer pico de esta. Debido a fuentes de investigación, se concluye que en los primeros meses de los años, la principal causante del aumento de accidentes son las lluvias esporadicas que hacen que el asfalto sea resbalizo, sumado a esto, las negligencias de conductores y altas velocidades.
Por otra parte, al observar los últimos años tenidos en cuenta en la gráfica. Se analiza que los accidentes de tránsito presentar un alto aumento y como método de concientización no solo deben realizar campañas si no, tener un estricto seguimiento en las vías con cámaras de reducción de velocidad o una constante revisión de estos vehículos.

##**5. ¿Como se distrubuyen las víctimas fallecidas y lesionadas por año?. ¿Cómo sería la distribución de fallecidos y lesionados según el género en cada caso?**

In [ ]:
fig51.show()
fig5.show()
fig52.show()

**Análisis**

A partir de la gráfica se puede concluir que los hombres continúan siendo el pico en el tipo de evento fallecidos, esto se debe a que existe mayor proporción de conductores en el país de Guatemala. En este caso, se tienen alrededor de 8265 hombres fallecidos en todos los años teniendo en cuenta los diferentes periodos. Sumado a esto, se tiene como mayor pico el 2016 con un total de 2000 casos de fallecidos.

##**6. Por día de la semana, ¿cómo se distribuyen los accidentes de tránsito? ¿Cómo se distribuyen los fallecidos y lesionados?**

In [ ]:
fig6.show()
fig66.show()

**Análisis**

La mayor cantidad de accidentes se presenta durante los fines de semana, siendo ligeramente mayor los domingos sobre los sábados.

La cantidad de lesionados siempre es mayor que los fallecidos sin importar el día. De igual forma tanto los fallecidos como los lesionados se presentan en mayor cantidad los fines de semana.

##**7. ¿Cómo es la proporción de internados en centros médicos? ¿Cómo cambia según el género? ¿Cómo afecta el tipo de evento la cantidad de internados?**

In [ ]:
fig7.show()
fig77.show()
fig777.show()

**Análisis**

Del 100% de los accidentados alrededor del 77.1% son internados o llevados a centros médicos, el 22.4% no requieren ser atendidos en centros médicos y el porcentaje restante, 0.44%, se desconoce la información.

De ese 77.% de atendidos en centros médicos, 31375 son hombres y 13975 son mujeres.

Además, el mayor tipo de accidente de tránsito que remite personas a centros médicos en la colisión, seguido por atropello y choque de vehículos. Hay que tener presente que son el tipo de accidente más frecuente en las vías.

La cantidad de lesionados siempre es mayor que los fallecidos sin importar el día. De igual forma tanto los fallecidos como los lesionados se presentan en mayor cantidad los fines de semana.

##**8. ¿ Cantidad de personas involucradas en eventos de transito en estado de embriaguez por mes y por departamento?**

In [ ]:
fig8.show()

**Análisis**

Se observa que el departamento donde mas se presenta accidentes de personas en estado de embriaguez es en el departameno de guatemala y en el mes de julio el cual es fecha de vacaciones.

##**9.¿Cómo se comporta el número de fallecidos, ebrios y colisiones según el mes?**

In [ ]:
fig9.show()

**Análisis**

Se observa que los fallecidos, las personas en estado de embriaguez y las colisiones aumentan en el mes Diciembre lo que puede ser explicado por la temporada alta ya que el nivel de transito se incrementa ademas, es el mes donde se presentan mayor festividades.

Respecto al comportamiento de cada uno la mayor cantidad de fallecidos y colisiones se presentan en los meses de Marzo y Diciembre. La temporada de marzo es posible explicarla ya que de acuerdo a la investigacion esta es la epoca donde mas llueve en guatemala.

Respecto a la cantidad de conductores ebrios se observa picos en los meses de marzo, julio, septiembre y diciembre. La temporada alta comprende los meses de Diciembre a Abril, además en marzo se celebra el dia de la mujer lo cual es una fecha muy importante en el pais. 
En julio se presenta las vacaciones y tienen la festividad del dia mundial de la población, por ultimo en septiembre se celebra el día de la independencia, lo que puede siginificar un mayor numero de festividades respecto a los demas meses.



# **Indicadores**

In [41]:
p4= vi[['color_veh', 'hora_ocu']]
p4= p4.groupby('color_veh')[['hora_ocu']].count().reset_index().rename(columns={'hora_ocu': 'cantidad'}).sort_values('cantidad', ascending= False)

x1 = dicc.loc[dicc['Variable'] == 'color_veh'][['Codigo', 'Valor']].rename(columns={'Codigo': 'color_veh', 'Valor':'color'}).drop(columns=['color_veh'])
x1['color'] = x1['color'].replace(['Ignorado'], 'Sin dato')
p4 = pd.merge(p4, x1, how='left', on='color_veh')
p4

KeyError: ignored

In [31]:
vi['edad_per'].unique()

array([ 24,  19,  32,  18, 999,  20,  23,  29,  21,  28,  33,  25,  22,
        30,  82,  26,  16,  45,  47,  34,  66,  27,  58,  64,  35,  48,
        38,  40,  53,  61,  41,  37,  44,  14,  42,  77,  65,  49,  51,
        39,  54,  50,  36,  43,  31,  59,  17,  56,  63,  52,  60,  46,
        72,  15,  79,  68,  12,  55,  73,  57,  74,  78,  67,  76,  62,
        70,  13,  71,  85,  69,  75,  80,  88,  81,  84,   9,  87,  11,
        10,  89,  90,  86,  95,   6,  94,  92,   8,  83,   3])

In [32]:

vi = vi.loc[vi['edad_per'] != 999]
vi['edad_per'].unique()


array([24, 19, 32, 18, 20, 23, 29, 21, 28, 33, 25, 22, 30, 82, 26, 16, 45,
       47, 34, 66, 27, 58, 64, 35, 48, 38, 40, 53, 61, 41, 37, 44, 14, 42,
       77, 65, 49, 51, 39, 54, 50, 36, 43, 31, 59, 17, 56, 63, 52, 60, 46,
       72, 15, 79, 68, 12, 55, 73, 57, 74, 78, 67, 76, 62, 70, 13, 71, 85,
       69, 75, 80, 88, 81, 84,  9, 87, 11, 10, 89, 90, 86, 95,  6, 94, 92,
        8, 83,  3])

In [33]:
bins = [0, 5, 11, 18, 26, 59, 100]
names=['0-5','6-11','12-18','18-26','27-59','>60']
vi['edad_per']= pd.cut(vi['edad_per'], bins, labels= names)
vi['edad_per'].unique()

['18-26', '27-59', '12-18', '>60', '6-11', '0-5']
Categories (6, object): ['0-5' < '6-11' < '12-18' < '18-26' < '27-59' < '>60']

In [34]:
vi= vi.groupby(['edad_per'])[['num_corre']].count().reset_index()
vi

edad_per  num_corre
0      0-5          1
1     6-11         12
2    12-18       2465
3    18-26      14876
4    27-59      27409
5      >60       1757

In [36]:
fig = px.bar(vi, x='edad_per', y='num_corre', width=380, height =370)

fig.update_layout(xaxis_title="<b>Atacante<b>",
                  yaxis_title="<b>Edades<b>",
                  template = 'simple_white',
                  plot_bgcolor='rgba(0,0,0,0)')

fig.show()

In [37]:
ht.columns

Index(['num_corre', 'ano_ocu', 'mes_ocu', 'dia_ocu', 'dia_sem_ocu', 'hora_ocu',
       'g_hora', 'g_hora_5', 'depto_ocu', 'mupio_ocu', 'zona_ocu', 'tipo_veh',
       'marca_veh', 'color_veh', 'modelo_veh', 'g_modelo_veh', 'tipo_eve',
       'fecha'],
      dtype='object')

In [86]:
ht4=ht.groupby('mes_ocu')[['num_corre']].count().reset_index().rename(columns={'num_corre': 'cantidad'})
ht4

mes_ocu  cantidad
0         1      3390
1         2      3504
2         3      3644
3         4      3189
4         5      3029
5         6      2949
6         7      3198
7         8      3299
8         9      3238
9        10      3388
10       11      3557
11       12      4104

In [82]:
ht4['num_corre'].sum()

KeyError: ignored

In [84]:
ht4['promedio']=ht4['cantidad'].apply(lambda x: x/40489*100).round(2)
ht4= ht4.drop(columns='cantidad')
ht4

mes_ocu  promedio
0         1      8.37
1         2      8.65
2         3      9.00
3         4      7.88
4         5      7.48
5         6      7.28
6         7      7.90
7         8      8.15
8         9      8.00
9        10      8.37
10       11      8.79
11       12     10.14

In [89]:
fig = px.funnel(ht4, x='mes_ocu', y='cantidad',color="mes_ocu")

fig.update_layout(
    
    legend_title ="Medios de transporte",
    template = 'simple_white',
    title_x = 0.5,
    plot_bgcolor='rgba(0,0,0,0)')
   
fig.show()
    
    